resources/js/calendar-widget.js

```diff
@@ -38,12 +40,12 @@ export default function calendarWidget({
                 locale: locale,
                 firstDay: firstDay,
                 dayMaxEvents: dayMaxEvents,
-                selectable: hasDateSelectContextMenu,
+                selectable: hasDateSelectContextMenu || dateSelectEnabled,
                 eventStartEditable: eventDragEnabled,
                 eventDurationEditable: eventResizeEnabled,
             };
 
-            if (hasDateClickContextMenu) {
+            if (hasDateClickContextMenu && !dateClickEnabled) {
```

Comment:
Please refactor this into an else if branch. Prioritize custom event -> contextMenu, so first check dateClickEnabled and after that context menu.

Reply:
Would this result into if else based on just one condition instead of the two bools ? How do you see this?

Reply:
Yes exactly.

Basically in the first if statement you would only check `dateClickEnabled` and in the else if statement you would check the `dateClickContextMenu`.

Same thing for the select event and select context menu.

It's easier to read / maintain and indicates more clearly which feature takes precedence over the other imo.

Reply:
I processed your feedback into an 'if else if'